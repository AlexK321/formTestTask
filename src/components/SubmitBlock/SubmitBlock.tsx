import React, { FC } from 'react';
import './SubmitBlock.less';
import { Button } from 'antd';

interface ParamTypes {
  recoveryTime: number;
}

const SubmitBlock: FC<ParamTypes> = ({ recoveryTime }) => {
  return (
    <div className="form-row">
      <Button type="primary" htmlType="submit" disabled={Boolean(recoveryTime > 0)}>
        Отправить заявку {recoveryTime > 0 && `через ${recoveryTime} сек.`}
      </Button>
      <p className="form-description">
        Нажимая на кнопку «Отправить заявку», вы соглашаетесь на обработку персональных данных
      </p>
    </div>
  );
};

export default SubmitBlock;
