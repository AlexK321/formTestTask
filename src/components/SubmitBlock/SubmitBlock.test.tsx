import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import SubmitBlock from './SubmitBlock';

describe('Component: SubmitBlock', () => {
  let component: ShallowWrapper;

  beforeEach(() => {
    component = shallow(<SubmitBlock SMSResendTimer={20} />);
  });

  it('match snapshot component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render title', () => {
    const consentDataProcessing =
      'Нажимая на кнопку «Отправить заявку», вы соглашаетесь на обработку персональных данных';
    const wrapper = component.find('.form-description');

    expect(wrapper.text()).toBe(consentDataProcessing);
  });
});
