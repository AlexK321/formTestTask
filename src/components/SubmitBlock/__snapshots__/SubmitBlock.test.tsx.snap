// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SubmitBlock match snapshot component 1`] = `
<div
  className="form-row"
>
  <Button
    disabled={true}
    htmlType="submit"
    type="primary"
  >
    Отправить заявку 
    через 20 сек.
  </Button>
  <p
    className="form-description"
  >
    Нажимая на кнопку «Отправить заявку», вы соглашаетесь на обработку персональных данных
  </p>
</div>
`;
